disease_gene/disease_associates_gene/generative_model_experiments/gen_model_benchmarking.py

```diff
@@ -0,0 +1,335 @@
+                          yaxis_tick_labels=lf_names, xaxis_tick_labels=lf_names,
+                          figsize=(10,8), colorbar=False, plot_title="Conflict Matrix")
+
+
+# The conflict matrix above shows how often label functions conflict with each other. The brighter the color the more conflict a label function has with another function. Ignoring the diagonals, there isn't many conflicts between functions except for the LF_DG_NO_CONCLUSION and LF_DG_ALLOWED_DISTANCE.
+
+# # Train the Generative Model
+
+# After visualizing the label functions and their associated properties, now it is time to work on the generative model. As with common machine learning pipelines, the first step is to find the best hyperparameters for this model. Using the grid search algorithm, the follow parameters were optimized: amount of burnin, strength of regularization, number of epochs to run the model.
+
+# ## Set the hyperparameter grid search
+
+# In[13]:
+
+
+regularization_grid = pd.np.round(pd.np.linspace(0.1, 6, num=25), 3)
+
+
+# ## What are the best hyperparameters for the conditionally independent model?
+
+# In[37]:
+
+
+L = convert_labels(label_matricies['train'].toarray(), 'plusminus', 'categorical')
+L_dev = convert_labels(label_matricies['dev'].toarray(), 'plusminus', 'categorical')
+L_test = convert_labels(label_matricies['test'].toarray(), 'plusminus', 'categorical')
+
+validation_data = list(zip([L[:,:7], L[:, :24], L], [L_dev[:,:7], L_dev[:, :24], L_dev]))
+test_data = list(zip([L[:,:7], L[:, :24], L], [L_test[:,:7], L_test[:, :24], L_test]))
+model_labels = ["Knowledge Bases (KB)", "KB+Text Patterns", "All"]
+
+
+# In[15]:
+
+
+model_grid_search = {}
+for model_data, model_label in zip(validation_data, model_labels):
+    
+    label_model = LabelModel(k=2, seed=100)
+    grid_results = {}
+    for param in regularization_grid:
+        label_model.train_model(model_data[0], n_epochs=1000, verbose=False, lr=0.01, l2=param)
+        grid_results[str(param)] = label_model.predict_proba(model_data[1])[:,0]
+        
+    model_grid_search[model_label] = pd.DataFrame.from_dict(grid_results)
+
+
+# In[16]:
+
+
+model_grid_aucs = {}
+for model in model_grid_search:
+    model_grid_aucs[model] = plot_curve(model_grid_search[model], candidate_dfs['dev'].curated_dsh, 
+                               figsize=(16,6), model_type='scatterplot', plot_title=model, metric="ROC", font_size=10)
+
+
+# In[17]:
+
+
+model_grid_auc_dfs = {}
+for model in model_grid_aucs:
+    model_grid_auc_dfs[model] = (
+        get_auc_significant_stats(candidate_dfs['dev'], model_grid_aucs[model])
+        .sort_values('auroc', ascending=False)
+    )
+    print(model)
+    print(model_grid_auc_dfs[model].head(5))
+    print()
+
+
+# # Final Evaluation on Held out Hand Labeled Test Data
+
+# In[39]:
+
+
+dev_model_df = pd.DataFrame()
+for best_model, model_data, model_label in zip([1.083, 2.067, 1.575], validation_data, model_labels):
```

Comment:
What are these floats in your zip function?

Reply:
These floats are the best hyper parameters selected for the different models to get the final evaluation on the test set.